www/kasj08/semestralka2/scripts/main.js

```diff
@@ -0,0 +1,504 @@
+            }
+            if (artists.length < 1) {
+                ShowError('No albums', 'Cannot get any album, because you are not following any artist.');
+                return;
+            }
+
+            if (json.artists.next) {
+                // pokud existuje další stránka seznamu umělců, získám další seznam
+                for (let index = 0; index < artists.length; index++) {
+                    // projde získané umělce a získá jejich alba
+                    GetArtistAlbums(artists[index], false);
+                }
+                GetUserArtists(json.artists.next);
+            }
+            else {
+                // neexistuje další stránka umělců
+                lastAlbumsCurrent = 0;
+                lastAlbumsCount = artists.length;
+                for (let index = 0; index < artists.length; index++) {
+                    GetArtistAlbums(artists[index], true);
+                }
+            }
+        });
+}
+
+function ShowAlbums() {
+    // jedná se o poslední stránku umělců
+
+    // přičtu aktuální počet úspěšně získaných posledních albumů
+    lastAlbumsCurrent++;
+    if (lastAlbumsCount != lastAlbumsCurrent) {
+        // ještě jsem neprošel všechny interprety
+        return;
+    }
+
+    // úspěšně jsem získal ze spotify interprety z poslední stránky
+    // zobrazím alba
+    if (!userAlbums) {
+        ShowError('No albums', 'Cannot get any album, because you are following only artists without albums.');
+        return;
+    }
+    if (userAlbums.length < 1) {
+        ShowError('No albums', 'Cannot get any album, because you are following only artists without albums.');
+        return;
+    }
+
+    // zobrazím roky vydaných alb umělců v menu
+    $('.title').text('All released albums');
+    AddMenuYears();
+    // získám rok a měsíc nejnovějšího alba
+    var date = userAlbums[0].release.split('-');
+    var year = date[0];
+    var month = date[1];
+    // zobrazení v menu
+    $('#' + year).addClass("current-year");
+    $('#m' + year).addClass("selected-month");
+    $('#' + date[0] + '-' + date[1]).addClass("current-month");
+    // zobrazení albumů
+    ViewAlbums(year, month);
+    $('.message').remove();
+}
+
+/* získá z api json seznam alb jednotlivých umělců */
+function GetArtistAlbums(artist, lastCheck) {
+    if (!artist) {
+        ShowError('Error', 'Failed to get artist albums.');
+        return;
+    }
+
+    $('.message').text('Please wait: Getting albums from artist ' + artist.name + '...');
+    var fetchUrl = 'https://api.spotify.com/v1/artists/' + artist.id + '/albums?offset=0&limit=50&include_groups=album&market=' + userCountry;
+
+    fetch(fetchUrl, options)
+        .then(response => response.json())
+        .then(json => {
+            // možné chyby
+            if (json.error) {
+                if (json.error.status == 429) {
+                    // chyba api - moc dotazů
+                    setTimeout(GetArtistAlbums(artist, lastCheck), 100);
```

Comment:
uz jsi nekdy dostal tenhle error?

Reply:
ano, dělal jsem si testovací účet, který získá kolem 5000 alb a tam se to stává
navíc myslím, že pokud by aplikaci využívalo více uživatelů najednou, tak by to bylo častější